src/io_helpers/mod.rs

```diff
@@ -4,16 +4,30 @@ use std::io::Read;
 
 use actix_web::HttpResponseBuilder;
 use anyhow::Result;
+use dashmap::DashMap;
 
 // this should be something else
 // probably inside the submodule of the http router
 #[inline]
-pub fn apply_headers(response: &mut HttpResponseBuilder, headers: &HashMap<String, String>) {
-    for (key, val) in (headers).iter() {
+pub fn apply_hashmap_headers(
+    response: &mut HttpResponseBuilder,
+    headers: &HashMap<String, String>,
+) {
+    for (key, val) in headers.iter() {
         response.insert_header((key.clone(), val.clone()));
     }
 }
 
+#[inline]
+pub fn apply_dashmap_headers(
+    response: &mut HttpResponseBuilder,
+    headers: &DashMap<String, String>,
+) {
+    for h in headers.iter() {
+        response.insert_header((h.key().clone(), h.value().clone()));
+    }
+}
+
```

Comment:
This is very smart! Good job! 👍

Reply:
Thanks ;) This avoids merging the headers and then applying them, which creates an unnecessary copy of both headers list. I wanted to make only one function that works with either a hashmap or a dashmap but it doesn't seem to be easy